lib/vagrant/action/builtin/cloud_init_setup.rb

```diff
@@ -0,0 +1,130 @@
+require 'mime'
+require 'tmpdir'
+
+module Vagrant
+  module Action
+    module Builtin
+      class CloudInitSetup
+        TEMP_PREFIX = "vagrant-cloud-init-iso-temp-".freeze
+
+        def initialize(app, env)
+          @app    = app
+          @logger = Log4r::Logger.new("vagrant::action::builtin::cloudinit::setup")
+        end
+
+        def call(env)
+          machine = env[:machine]
+
+          user_data_configs = machine.config.vm.cloud_init_configs
+                                                  .select { |c| c.type == :user_data }
+
+          if !user_data_configs.empty?
+            user_data = setup_user_data(machine, env, user_data_configs)
+            meta_data = { "instance-id": "i-#{machine.id.split('-').join}" }
+
+            write_cfg_iso(machine, env, user_data, meta_data)
+          end
+
+          # Continue On
+          @app.call(env)
+        end
+
+        # @param [Vagrant::Machine] machine
+        # @param [Vagrant::Environment] env
+        # @param [Array<#VagrantPlugins::Kernel_V2::VagrantConfigCloudInit>] user_data_cfgs
+        # @return [MIME::Text] user_data
+        def setup_user_data(machine, env, user_data_cfgs)
+          machine.ui.info(I18n.t("vagrant.actions.vm.cloud_init_user_data_setup"))
+
+          text_cfgs = []
+          user_data_cfgs.each do |cfg|
+            text_cfgs << read_text_cfg(machine, cfg)
+          end
+
+          user_data = generate_cfg_msg(machine, text_cfgs)
+          user_data
+        end
+
+        # Reads an individual cloud_init config and stores its contents and the
+        # content_type as a MIME text
+        #
+        # @param [Vagrant::Machine] machine
+        # @param [VagrantPlugins::Kernel_V2::VagrantConfigCloudInit] cfg
+        # @return [MIME::Text] text_msg
+        def read_text_cfg(machine, cfg)
+          if cfg.path
+            text = File.read(Pathname.new(cfg.path).expand_path(machine.env.root_path))
+          else
+            text = cfg.inline
+          end
+
+          # Note: content_type must remove the leading `text/` because
+          # the MIME::Text initializer hardcodes `text/` already to the type.
+          # We assume content_type is correct due to the validation step
+          # in VagrantConfigCloudInit.
+          content_type = cfg.content_type.split('/')

```

Comment:
`content_type = cfg.content_type.split("/", 2).last`

Will ensure it only splits once if there are multiple slashes (there shouldn't be) and still returns the value if the `text/` is not provided (which it shouldn't be, but provides safety for if it is not).